### DISCUSSION TITLE
Bug in ignore whitespace ?

### DISCUSSION THREAD

**Discussion by remfan77** [WinMerge/winmerge#1881] [Q&A]
Hello,
I do not understand how WinMerge ignore whitespace works.

I use the option **ignore white space**. 

I have two folder A and B.
Each folder A and B contains 2 files,

1. FILE.txt
2. SPI.c

Ignoring whitespace folder A and B should be the same.

[WINMERGE-BUG-WHITESPACE.zip](https://github.com/WinMerge/winmerge/files/11599676/WINMERGE-BUG-WHITESPACE.zip)

Why I see

![WINMERGE-BUG-WHITESPACE](https://github.com/WinMerge/winmerge/assets/52342624/35d7c47d-29de-4004-8528-5727015ebd25)

But If I click on SPI.c WinMerge correctly tell me SPI.c are the same.

Can someone explain the reason ?
Is it a bug or I miss something ?

Best regards,
Paolo

**Comment by sdottaka** [comment]
You probably have the following settings

Whitespaces: Ignore change or Ignore all
Ignore blank lines: enabled
Compare method: Quick Contents

If you set Compare method to Full Contents, you will get the result you expected.

It's most likely a Quick Contents compare method bug.

**Comment by remfan77** [comment]
Thanks for you quick response.

Do you mean Diff Algo (experimental) ?
In WinMerge 2.16.30 I have
- default
- min
- patience
- histogram
- none

Tried all, but I have always the same problem.

**Reply by sdottaka** [reply]
Compare method can be changed as follows.

![image](https://github.com/WinMerge/winmerge/assets/98126/c6c98aba-1c96-4f76-b495-a5d51731ee0e)

**Comment by remfan77** [comment]
Thanks a lot @ sdottaka.
I didn't know about this menu.
Best regards
Paolo

**Reply by sdottaka** [reply]
Thank you for your report. This issue has been fixed in commit [e234c3c](https://github.com/WinMerge/winmerge/commit/e234c3c4743be775c727dd3f631cd59531197d03).

**Comment by remfan77** [comment]
Thank you for support and the fix.
